crates/ironrdp-rdpdr-native/src/nix/simple.rs

```diff
@@ -0,0 +1,765 @@
+                    let res = RdpdrPdu::DeviceCreateResponse(DeviceCreateResponse {
+                        device_io_reply: io_response,
+                        file_id,
+                        information: Information::empty(),
+                    });
+                    return Ok(vec![SvcMessage::from(res)]);
+                }
+                if req_inner.create_options.bits() & CreateOptions::FILE_NON_DIRECTORY_FILE.bits() != 0 {
+                    warn!("create file,but it is directory");
+                    let io_response = DeviceIoResponse::new(req_inner.device_io_request, NtStatus::UNSUCCESSFUL);
+                    let res = RdpdrPdu::DeviceCreateResponse(DeviceCreateResponse {
+                        device_io_reply: io_response,
+                        file_id,
+                        information: Information::empty(),
+                    });
+                    return Ok(vec![SvcMessage::from(res)]);
+                }
+                //这个我们先不执行，我们看看能不能统一
+                //我们可以统一到文件打开的条件中去
+            } else if req_inner.create_options.bits() & CreateOptions::FILE_DIRECTORY_FILE.bits() != 0 {
+                warn!("create directory,but it is file");
+                let io_response = DeviceIoResponse::new(req_inner.device_io_request, NtStatus::NOT_A_DIRECTORY);
+                let res = RdpdrPdu::DeviceCreateResponse(DeviceCreateResponse {
+                    device_io_reply: io_response,
+                    file_id,
+                    information: Information::empty(),
+                });
+                return Ok(vec![SvcMessage::from(res)]);
+            }
+        }
+        Err(_) => {
+            if req_inner.create_options.bits() & CreateOptions::FILE_DIRECTORY_FILE.bits() != 0 {
+                warn!("create directory,but it is file");
+                if (req_inner.create_disposition == CreateDisposition::FILE_CREATE
+                    || req_inner.create_disposition == CreateDisposition::FILE_OPEN_IF)
+                    && std::fs::create_dir_all(path.as_str()).is_ok()
+                {
+                    let mut fs = std::fs::OpenOptions::new();
+                    match fs.read(true).open(&path) {
+                        Ok(file) => {
+                            debug!("file_id:{},path:{}", file_id, path);
+                            backend.file_map.insert(file_id, file);
+                            backend.file_path_map.insert(file_id, path.clone());
+                            return make_create_drive_resp(
+                                req_inner.device_io_request,
+                                req_inner.create_disposition,
+                                file_id,
+                            );
+                        }
+                        Err(e) => {
+                            warn!("open file dir error:{:?}", e);
+                            //return by downside
```

Comment:
style: See [this](https://github.com/Devolutions/IronRDP/blob/abc0ee0617fcdcda3f0ace3ffd292251979d1910/STYLE.md#inline-code-comments-are-proper-sentences)

Reply:
ok, i modify according to our style